runtime/util/annhelp.c

```diff
@@ -139,3 +143,122 @@ skipAnnotationElement(J9ROMConstantPoolItem const * constantPool, U_8 const *dat
 	*pIndex = index;
 	return -1;
 }
+
+/**
+ * Check if a field contains the specified Runtime Visible annotation.
+ *
+ * @param clazz The class the field belongs to.
+ * @param cpIndex The constant pool index of the field.
+ * @param annotationName The name of the annotation to check for.
+ * @return TRUE if the annotation is found, FALSE otherwise.
+ */
+BOOLEAN
+fieldContainsRuntimeAnnotation(J9VMThread *currentThread, J9Class *clazz, UDATA cpIndex, J9UTF8 *annotationName)
```

Comment:
The doc (and probably function name) should indicate that the field ref must be resolved. You may also want to add some asserts to that end.